### PR TITLE
Fix version check

### PR DESCRIPTION
**Pull request by raskchanky**
This should fix failures like this https://app.circleci.com/pipelines/github/hashicorp/vault-enterprise/11490/workflows/d2540ff5-7327-4253-808d-878bd2853483/jobs/362948/tests#failed-test-1 on enterprise

### CODE REVIEW

File: vault/version_store.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (c *Core) storeVersionTimestamp(ctx context.Context, version string, timest
 // FindOldestVersionTimestamp searches for the vault version with the oldest
 // upgrade timestamp from storage. The earliest version this can be is 1.9.0.
 func (c *Core) FindOldestVersionTimestamp() (string, time.Time, error) {
-	if len(c.versionTimestamps) == 0 {
+	if c.versionTimestamps == nil {
```
Comment (HridoyRoy) [inline vault/version_store.go:73]: I think these don't need to be nil checks right? Checking the length for a nil value should result in 0.

Comment (raskchanky) [reply vault/version_store.go:73]: You're right, but in this case, having a zero length struct is not an error condition. We should only be erroring if the struct is actually nil.

Comment (HridoyRoy) [reply vault/version_store.go:73]: Ah, yes that makes sense!

**Review by HridoyRoy — APPROVED** [review]
Lgtm!

**Pull request merged**